### DISCUSSION TITLE
Allow editing of kWh "Used" in Charges

### DISCUSSION THREAD

**Discussion by petariga** [teslamate-org/teslamate#684] [Ideas]
**What would you like to be added**:
Allow users to edit kWh in "Used" in Charges similarly how you can set or edit the "Charge Cost" (per each charging session).

**Why is this needed**:
When I charge on public chargers they count kWh according to their measurement and this figure is usually 10-20% higher than the amount of kWh which Tesla receives and states in the infotainment (I assume this is the value "Added" in TeslaMate). So far the TeslaMate states that there is only about 1% of lost between "Used" and "Added" (99% efficiency) when I charge on public CCS chargers, however this is not correct. I assume that Tesla is not passing the figures from the external CCS charger correctly (or they might not get them, I don't know the standard).
I am billed according to the measurement from the owner of the charger, not from Tesla infotainment, therefore I would like to keep the track of the real numbers.

**Additional context**:
I am fairly new to TeslaMate, so might not be familiar with all the features it offers. It is a great piece of software, I really appreciate it.

**Comment by adriankumpf** [comment]
> I assume that Tesla is not passing the figures from the external CCS charger correctly. 

The "Used" value is actually calculated by TeslaMate based on the charge data (`       charger_actual_current * charger_voltage * phases / 1000.0`). It might therefore deviate from what the charger indicates or what is billed. However, It gives an idea of the losses during charging. But this depends on where the measurement is taken. In this case we use the data measured by the car and not at the charger.

**Comment by petariga** [comment]
Thank you for your explanation. However, if they are calculated in TeslaMate not "correctly" (they do not reflect the actual electricity used during the charging), can you let the users override (edit/set) this value manualy? For me, as a user, it makes much more sense to keep the track of the real "consumed/used" kWh supplied by the charger (which is always shown at the end of charging on the charger) and billed to me, than a figure which is somewhere on the half way to reality...   Or maybe another attribute/value in the database would sort this out as well, however it might bring more confusion to users...
